examples/tutorials/core_api_mnist/model_def_checkpoints.py

```diff
@@ -0,0 +1,184 @@
+                        help='quickly check a single pass')                    
+    parser.add_argument('--seed', type=int, default=1, metavar='S',
+                        help='random seed (default: 1)')
+    parser.add_argument('--log-interval', type=int, default=10, metavar='N',
+                        help='how many batches to wait before logging training status')
+    parser.add_argument('--save-model', action='store_true', default=True,
+                        help='For Saving the current Model')
+  
+    args = parser.parse_args()
+    use_cuda = not args.no_cuda and torch.cuda.is_available()
+    use_mps = not args.no_mps and torch.backends.mps.is_available()
+
+    # NEW: if checkpoint exists, load and save as model state prior to resuming training
+    info = det.get_cluster_info()
+    assert info is not None, "this example only runs on-cluster"
+    latest_checkpoint = info.latest_checkpoint
+    if latest_checkpoint is not None: 
+        with core_context.checkpoint.restore_path(latest_checkpoint) as path:
+            model = load_state(path)
+
+    torch.manual_seed(args.seed)
+
+    if use_cuda:
+        device = torch.device("cuda")
+    elif use_mps:
+        device = torch.device("mps")
+    else:
+        device = torch.device("cpu")
+
+    train_kwargs = {'batch_size': args.batch_size}
+    test_kwargs = {'batch_size': args.test_batch_size}
+    if use_cuda:
+        cuda_kwargs = {'num_workers': 1,
+                       'pin_memory': True,
+                       'shuffle': True}
+        train_kwargs.update(cuda_kwargs)
+        test_kwargs.update(cuda_kwargs)
+
+    transform=transforms.Compose([
+        transforms.ToTensor(),
+        transforms.Normalize((0.1307,), (0.3081,))
+        ])
+    dataset1 = datasets.MNIST('../data', train=True, download=True,
+                       transform=transform)
+    dataset2 = datasets.MNIST('../data', train=False,
+                       transform=transform)
+    train_loader = torch.utils.data.DataLoader(dataset1,**train_kwargs)
+    test_loader = torch.utils.data.DataLoader(dataset2, **test_kwargs)
+
+    model = Net().to(device)
+    optimizer = optim.Adadelta(model.parameters(), lr=args.lr)
+
+    scheduler = StepLR(optimizer, step_size=1, gamma=args.gamma)
+    for epoch in range(1, args.epochs + 1):
+        len_trainset = len(train_loader)
+        train(args, model, device, train_loader, optimizer, epoch, core_context)
+        test(args, model, device, test_loader, epoch, core_context, len_trainset)
+        scheduler.step()
+        
+        # NEW: save checkpoint
+        if args.save_model:
+            
+            checkpoint_metadata_dict = {
+                'steps_completed': len(train_loader),
+            }
+            
+        with core_context.checkpoint.store_path(checkpoint_metadata_dict) as (path, storage_id):
+            torch.save(model.state_dict(), str(path) + ('/checkpoint.pt'))
```

Comment:
can you simplify `str(path) + ('/checkpoint.pt')` to just `path / "checkpoint.pt"`?  If `torch.save` allows it, it looks a lot neater I think.